File: Q_728.cpp
class Solution {
public:
    vector<int> ret;
    
    void check_num(int a){
        int temp=a,i;
        if(a==0)
            return;
        while(a>0){
            i = a%10;
            if(i==0)
                break;
            if(temp%i!=0)
                break;
            a=a/10;
        }
        if(a==0)
            ret.push_back(temp);
        
    }
    vector<int> selfDividingNumbers(int left, int right) {
        int i;
        for(i=left;i<=right;i++)
            check_num(i);
        return ret;
    }
};

File: python/Q_728.py
class Solution:
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        mylist =[]
        
        for i in range(left,right+1):
            num = i
            temp = i
            while num>0:
                digit=num%10
                if digit==0 or temp%digit!=0:
                    break
                num = num//10
            if num == 0:
                mylist.append(temp)
        return mylist
